### PR TITLE
Updates/sgb motus support

### PR DESCRIPTION
**Pull request by danielpodlesny**

### CODE REVIEW

File: samestr/summarize/summarize.py [hunk 1]
```diff
@@ -59,19 +53,15 @@ def summarize(args):
...
 
     # write output
-    mp_taxonomy.to_csv(args['output_dir'] +
-                       '/mp_taxonomy.tsv', sep='\t', index=False)
-    mp_species.drop(['kingdom', 'phylum', 'class', 'order', 'family', 'genus'], axis=1).to_csv(
-        args['output_dir'] + '/mp_species.tsv', sep='\t', index=False)
-    mp_counts.to_csv(args['output_dir'] +
-                     '/mp_counts.tsv', sep='\t', index=False)
+    taxon_counts.to_csv(args['output_dir'] +
+                     '/taxon_counts.tsv', sep='\t', index=False)
```
Comment (cschu) [inline samestr/summarize/summarize.py:64]:
```suggestion
    taxon_counts.to_csv(os.path.join(args['output_dir'],
                     'taxon_counts.tsv', sep='\t', index=False)
```

If you use `os.path.join`, you don't need to manually attach `/` everywhere

File: samestr/summarize/read_taxonomic_profiles.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+    return profile.set_index('clade')[profile.columns[1]].to_dict()
+
+
+def binarize_taxonomic_profile(df):
+    """
+    Take a taxonomic profile dataframe and return a binary dataframe
+    """
+    # convert to binary > numeric
+    return df.applymap(lambda x: 1 if x != 0 else 0)
```
Comment (cschu) [inline samestr/summarize/read_taxonomic_profiles.py:180]:
```suggestion
    return df.applymap(lambda x: int(bool(x)))
```
i think

File: samestr/filter/filter_freqs.py [hunk 3]
```diff
@@ -298,17 +327,26 @@ def filter_freqs(args):
     if args['delete_pos']:
         x = np.delete(x, list(removed_pos), axis=1)
 
+        # Save remaining positions to file
+        remaining_pos = sorted(initial_pos - removed_pos)
+        assert len(remaining_pos) == x.shape[1], "The number of remaining positions does not match the modified array shape."
+
+        with open(output_name + '.pos.txt', 'w') as file:
+            txt = '\n'.join([str(pos) for pos in remaining_pos])
+            file.write(txt)
```
Comment (cschu) [inline samestr/filter/filter_freqs.py:336]: `file` is a reserved keyword - it works, but should not be used

**Review by cschu — COMMENTED** [review]
- a lot of `+`-based string concatenation and manual path manipulation

File: samestr/utils/file_mapping.py [hunk 4]
```diff
@@ -131,3 +134,32 @@ def get_uniform_extension(files, accepted_extensions):
...
+    for cut in cuts:
+        if len(name) > prev_cut:
+            segment = name[prev_cut:cut] if len(name) >= cut else name[prev_cut:]
+            segments.append(segment)
+            prev_cut = cut
+        else:
+            break
+
+    # Combine the segments to form the path
+    pseudo_path = '/db_markers/' + '/'.join(segments) + '/'
```
[the remark this answers is not in the document]
Comment (danielpodlesny) [reply samestr/utils/file_mapping.py]: done

**Pull request merged**